eva/udfs/contrib/decorators/ish/udf_decorators.py

```diff
@@ -0,0 +1,121 @@
+# coding=utf-8
+# Copyright 2018-2022 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from eva.udfs.contrib.decorators.ish.io_descriptors.eva_arguments import EvaArgument
+from eva.utils.errors import TypeException
+
+
+# decorator for the setup function. It will be used to set the cache, batching and
+# udf_type parameters in the catalog
+def setup(use_cache: bool, udf_type: str, batch: bool):
+    def inner_fn(arg_fn):
+        print("Cache is set: ", use_cache)
+        print("batching is set: ", batch)
+
+        def wrapper(*args, **kwargs):
+
+            # TODO set the batch and caching parameters. update in catalog
+
+            # calling the setup function defined by the user inside the udf implementation
+            arg_fn(*args, **kwargs)
+
+        tags = {}
+        tags["cache"] = use_cache
+        tags["udf_type"] = udf_type
+        tags["batching"] = batch
+        wrapper.tags = tags
+        return wrapper
+
+    return inner_fn
+
+
+def forward(input_signature: EvaArgument, output_signature: EvaArgument):
```

Comment:
Probably need to rename `input_signature` to `inputs` and `output_signature` to `outputs`.